fossas/fossa-cli
# Prevent infinite recursion from Pom file property interpolation

# Overview

This address an issue where Maven static analysis hangs do to infinite recursion when pom file analysis attempts to interpolate a dependencies version property from a parent block (a parent pom or `dependencyManagement`). 

The issue arises when there is a block like the following `<junit5.version>${junit5.version}</junit5.version>`. 

Property interpolation works as follows:
- A dependency has the version `${junit5.version}`
- We clean (`splitTheProperty`) the maven property up to give us `junit5.version`
- We find a reference for `junit5.version` in a parent pom or `dependencyManagement` section
- That block looks like `<junit5.version>${junit5.version}</junit5.version>`
- We clean (`splitTheProperty`) the maven property up to give us `junit5.version`
- We find a reference for `junit5.version` in a parent pom or `dependencyManagement` section
- ...

## Acceptance criteria

- The user's project can successfully scan
- Infinite loops are no longer possible in the Pom file analysis

## Testing plan

Method 1
- Build the binary and ensure it can successfully analyze the user project in the Zendesk ticket

Method 2
- Add tests to ensure an infinite loop is handled
- Ensure they pass with the changes made
- Ensure they fail without the changes made

## Risks

I really can't think of any since the only this prevents is an infinite loop.

## References

https://fossa.zendesk.com/agent/tickets/6791

## Checklist
